Implemented Search using algolia

zzkzzzz opened the pull request:
Implemented Search frontend with algolia using algolia's "Vue Instant Search" component

zzkzzzz committed:
modified:   package-lock.json
	modified:   package.json
	modified:   src/main.js
	modified:   src/views/Explore.vue
	modified:   src/views/Favourites.vue
	src/components/InfiniteHits.vue

Jaimeloeuf commented on src/views/Explore.vue:
```diff
@@ -1,10 +1,192 @@
 <script>
+import algoliasearch from "algoliasearch/lite";
+import "instantsearch.css/themes/algolia-min.css";
+import AppInfiniteHits from "../components/InfiniteHits";
```
Missing component

zzkzzzz committed:
new file:   src/components/InfiniteHits.vue
added missing file

Jaimeloeuf commented on src/views/Explore.vue:
```diff
@@ -1,10 +1,192 @@
+  padding: 0;
+}
+
+body {
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica,
+    Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
+}
```
Possible to remove font to optimize it?

Jaimeloeuf commented on src/views/Favourites.vue:
```diff
@@ -80,13 +84,15 @@ import { mapGetters, mapActions } from "vuex";
 };
 </script>
+
+<style scoped></style>
```
remove empty scoped styles tag

Jaimeloeuf commented:
Remove trailing commas

Jaimeloeuf commented on src/views/Favourites.vue:
```diff
@@ -22,7 +22,11 @@
+          <v-img
+            id="class-image"
+            :src="clas.pictureSources[0]"
+            :aspect-ratio="16 / 9"
```
remove fixed aspect ratio of image

Jaimeloeuf commented on src/main.js:
```diff
@@ -6,6 +6,8 @@ import router from "./router";
 import store from "./store";
 import "./registerServiceWorker";
 import vuetify from "./plugins/vuetify";
+import InstantSearch from "vue-instantsearch";
+import VueObserveVisibility from "vue-observe-visibility";
```
Dont mount stuff onto Vue prototype with Vue.use in main.js to reduce initial load speed

Jaimeloeuf commented:
Limit infinite scroll to 20 for now

Jaimeloeuf commented:
Try to get user's current location and see if we can use it to search for nearby classes

zzkzzzz committed:
remove body font

zzkzzzz committed:
Removed empty scoped styles tag
Removed trailing commas

Jaimeloeuf committed:
Merge branch 'master' into search

Commit:
Display category hint in Explore page. Implement error handler if got
invalid search result.

zzkzzzz committed:
Display message if got no search result

zzkzzzz committed:
remove unnecessary import in main.js

zzkzzzz committed:
hide categories when there is search results
remove number by search card

zzkzzzz committed:
Remove algolia stuff from main.js
Rename custom explore to categories
Remove edits to settings.vue
Remove nodemon

zzkzzzz committed:
Add Filter Menu in Explore page

zzkzzzz committed:
Remove unnecessary comma
Change PopoverMenu design
Add in distance Slider to PopoverMenu

zzkzzzz committed:
Add RangeSlider with min and max to point filter

zzkzzzz committed:
Move Alogolia Search ID and Key to .env file.
Limit number of parter search result to 3.
